src/NanopassSharp/Patterns/EnumerablePattern.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NanopassSharp.Patterns;
+
+public sealed class EnumerablePattern : ITransformationPattern
```

Comment:
I feel like this is an aggregation pattern, you could actually refactor it a little to make it simpler. This works too, but we already have the concept in LINQ.